pkg/controller/elasticsearch/nodespec/lifecycle_hook.go

```diff
@@ -146,62 +146,69 @@ function supports_node_shutdown() {
 version=""
 if [[ -f "{{.LabelsFile}}" ]]; then
   # get Elasticsearch version from the downward API
-  version=$(grep "{{.VersionLabelName}}" {{.LabelsFile}} | cut -d '=' -f 2)
+  version=$(grep "{{.VersionLabelName}}" "{{.LabelsFile}}" | cut -d '=' -f 2)
   # remove quotes
   version=$(echo "${version}" | tr -d '"')
 fi
 
 # if ES version does not support node shutdown exit early
 if ! supports_node_shutdown "$version"; then
-  delayed_exit 
+  delayed_exit
 fi
 
 # setup basic auth if credentials are available
 if [ -f "{{.PreStopUserPasswordPath}}" ]; then
-  PROBE_PASSWORD=$(<{{.PreStopUserPasswordPath}})
-  BASIC_AUTH="-u {{.PreStopUserName}}:${PROBE_PASSWORD}"
+  PROBE_PASSWORD=$(<"{{.PreStopUserPasswordPath}}")
+  BASIC_AUTH=("-u" "{{.PreStopUserName}}:${PROBE_PASSWORD}")
 else
   # typically the case on upgrades from versions that did not have this script yet and the necessary volume mounts are missing
   log "no API credentials available, will not attempt node shutdown orchestration from pre-stop hook"
   delayed_exit
 fi
 
-ES_URL={{.ServiceURL}}
+ES_URL="{{.ServiceURL}}"
 
 log "retrieving node ID"
-retry 10 request -X GET "$ES_URL/_cat/nodes?full_id=true&h=id,name" $BASIC_AUTH
-if [ "$?" -ne 0 ]; then
-	error_exit "failed to retrieve node ID"
+if ! retry 8 request -X GET "${ES_URL}/_cat/nodes?full_id=true&h=id,name" "${BASIC_AUTH[@]}"
+then
+  error_exit "failed to retrieve nodes"
 fi
 
-NODE_ID=$(grep "$POD_NAME" "$resp_body" | cut -f 1 -d ' ')
+if ! NODE_ID="$(grep -oP "\K\w+(?= ${POD_NAME})" "${resp_body}")"
```

Comment:
This doesn't work when the node ID has `-`.

The simple thing I can think of to do the extraction is: `grep "${POD_NAME}" "${resp_body}" | awk '{print $1}'`.

I tested with this script:
```
#!/usr/bin/env bash

set -uo pipefail


req() {
  echo 'eb0GwvNWTkCuQngXptksnA pod-0
ACJTNKxESkes23TYiX6eZA pod-1
t1yM7brtQB256e-BVmM-Vw pod-2'
}

resp_body=tmpfile
req > $resp_body

test() {
  echo "-- test with POD_NAME=$POD_NAME"

  if ! NODE_ID="$(grep -oP "\K\w+(?= ${POD_NAME})" "${resp_body}")"; then
    echo no id
  else 
    echo id: $NODE_ID
  fi

  extract_node_id() {
    grep "${POD_NAME}" "${resp_body}" | awk '{print $1}'
  }

  if ! NODE_ID="$(extract_node_id  ${POD_NAME})"; then
    echo no id
  else 
    echo id: $NODE_ID
  fi

  echo
}

POD_NAME=pod-xyz
test

POD_NAME=pod-1
test

POD_NAME=pod-2
test
```

Reply:
Output:
```
> ./test.sh
-- test with POD_NAME=pod-xyz
no id 🟢
no id 🟢

-- test with POD_NAME=pod-1
id: ACJTNKxESkes23TYiX6eZA 🟢
id: ACJTNKxESkes23TYiX6eZA 🟢

-- test with POD_NAME=pod-2
id: Vw 🔴
id: t1yM7brtQB256e-BVmM-Vw 🟢
```

Reply:
Easy fix, \w is one word, use a different regex to match all before '<space><name>'

```shell
grep -oP ".*?(?=\ pod-1321)"
```

Reply:
```suggestion
if ! NODE_ID="$(grep -oP ".*?(?=\ ${POD_NAME})" "${resp_body}")"
```

Reply:
> Are there any more corner cases?

Sorry but I don't know.

Do you see an issue with using `grep | awk`? I would like to avoid using `grep --perl-regexp` if possible because regexes are often tricky.

Reply:
Nope, i don't see any issues, by habit i tend to make only one command and if possible ignore the pipe into other commands.
This leads to 2 commands that needs to function and checked.

Personal preference i think.

Reply:
Traditionally, I like regular expressions, but when it comes to filtering and at the same time extracting with a grouping, it starts to get a little trickier. Here, we prefer to prioritize readability for maintainability in case an operator is required to read this script.
So, yes please return to `grep | cut`.

Reply:
Yes i will push it tomorrow.